tests/gdx-tests/src/com/badlogic/gdx/tests/TiledMapObjectPropertyTest.java

```diff
@@ -0,0 +1,190 @@
+				MapProperties props = idPropMap.get(id);
+
+				switch (id) {
+				case 1:
+					test(props, 2, idPropMap);
+					test(props, 5, idPropMap);
+					test(props, 1, idPropMap);
+					break;
+				case 2:
+					test(props, 3, idPropMap);
+					test(props, 4, idPropMap);
+					break;
+				case 3:
+					test(props, 2, idPropMap);
+					break;
+				case 4:
+					test(props, 1, idPropMap);
+					break;
+				case 5:
+				case 6:
+					Iterator<String> propKeysIterator = props.getKeys();
+					ObjectSet<String> propKeys = new ObjectSet<>();
+					while (propKeysIterator.hasNext()) {
+						propKeys.add(propKeysIterator.next());
+					}
+					if (propKeys.size != 6) {
+						throw new RuntimeException("Object with id " + id + " should have six keys " + "but has " + propKeys);
+					}
+				}
+			}
+
+			for (ObjectMap.Entry<Integer, MapProperties> entry : idPropMap.entries()) {
+				int id = entry.key;
+				MapProperties props = entry.value;
+
+				System.out.println("Object with id " + id + " has \"object\" properties: ");
+
+				Iterator<String> propKeysIterator = props.getKeys();
+				Iterator<Object> propValuesIterator = props.getValues();
+
+				while (propKeysIterator.hasNext() && propValuesIterator.hasNext()) {
+					Object value = propValuesIterator.next();
+					String key = propKeysIterator.next();
+					if (!key.contains("Points_To_ID_")) {
+						continue;
+					}
+
+					if (value instanceof MapProperties) {
+						MapProperties nestedProps = (MapProperties)value;
+						value = "props of object with id " + nestedProps.get("id", Integer.class);
+					}
+
+					System.out.println("\t" + key + " = " + value);
+				}
+
+			}
+
+			Gdx.app.log("TiledMapObjectPropertyTest", "Successfully ran test!");
+		} catch (Exception e) {
+			Gdx.app.error("TiledMapObjectPropertyTest", "Failed to run test!", e);
+			e.printStackTrace();
+			error = true;
+		}
+	}
+
+	@Override
+	public void render () {
+		if (error) {
+			Gdx.app.error("TiledMapObjectPropertyTest", "Failed to run test!");
+			Gdx.app.exit();
+		}
+
+		ScreenUtils.clear(0.55f, 0.55f, 0.55f, 1f);
+		camera.update();
+		mapRenderer.setView(camera);
+		mapRenderer.render();
+
+		shapeRenderer.setProjectionMatrix(camera.combined);
+		batch.setProjectionMatrix(camera.combined);
+
+		shapeRenderer.setColor(Color.BLUE);
+		Gdx.gl20.glLineWidth(2);
+		MapLayer layer = map.getLayers().get("Objects");
+		for (MapObject mapObject : layer.getObjects()) {
+			if (!mapObject.isVisible()) continue;
+			if (mapObject instanceof RectangleMapObject) {
+				shapeRenderer.begin(ShapeRenderer.ShapeType.Filled);
+				Rectangle rectangle = ((RectangleMapObject)mapObject).getRectangle();
+				shapeRenderer.rect(rectangle.x, rectangle.y, rectangle.width, rectangle.height);
+				shapeRenderer.end();
+			}
+		}
+	}
+
+	@Override
+	public void dispose () {
+		map.dispose();
+		shapeRenderer.dispose();
+	}
+
+	private void test (MapProperties props, int idToObjProp, ObjectMap<Integer, MapProperties> idPropMap) {
```

Comment:
This would probably also need changing if you do switch the other `ObjectMap<Integer, MapProperties>` to use an IntMap.